livekit-agents/livekit/agents/worker.py

```diff
@@ -68,21 +68,35 @@ class WorkerPermissions:
 @dataclass
 class WorkerOptions:
     entrypoint_fnc: Callable[[JobContext], Coroutine]
+    """Entrypoint function that will be called when a job is assigned to this worker."""
     request_fnc: Callable[[JobRequest], Coroutine] = _default_request_fnc
+    """Inspect the request and decide if the current worker should handle it.
+
+    When left empty, all jobs are accepted."""
     prewarm_fnc: Callable[[JobProcess], Any] = _default_initialize_process_fnc
+    """A function to perform any necessary initialization before the job starts."""
     load_fnc: Callable[[], float] = _default_cpu_load_fnc
+    """Called to determine the current load of the worker. Should return a value between 0 and 1."""
     load_threshold: float = 0.8
+    """When the load exceeds this threshold, the worker will be marked as unavailable."""
     num_idle_processes: int = 3
+    """Number of idle processes to keep warm."""
     shutdown_process_timeout: float = 60.0
     initialize_process_timeout: float = 10.0
     permissions: WorkerPermissions = field(default_factory=WorkerPermissions)
+    """Permissions that the agent should join the room with."""
     worker_type: agent.JobType = agent.JobType.JT_ROOM
+    """Whether to spin up an agent for each room or publisher."""
     max_retry: int = MAX_RECONNECT_ATTEMPTS
     ws_url: str = "ws://localhost:7880"
+    """URL to connect to the LiveKit server. Uses LIVEKIT_URL from environment"""
     api_key: str | None = None
+    """API key to authenticate with LiveKit. Uses LIVEKIT_API_KEY from environment"""
     api_secret: str | None = None
+    """API secret to authenticate with LiveKit. Uses LIVEKIT_API_SECRET from environment"""
```

Comment:
thinking maybe we should actively discourage this use? we should let users shoot themselves in the foot but i think it should be made clearer

```python
"""API key to authenticate with LiveKit

For security reasons, it is recommended to use the LIVEKIT_API_KEY environmental variable instead."""

Reply:
users may want to pass in these vars for other reasons. i.e. their own secret vault. we'll leave security recommendations outside of the docstring themselves.